apps/minifront/src/fetchers/registry.ts

```diff
@@ -0,0 +1,28 @@
+import { getChainId } from './chain-id';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { getAssetMetadataById } from './assets';
+
+export const getIbcConnections = async () => {
+  const chainId = await getChainId();
+  if (!chainId) throw new Error('Could not fetch chain id');
+
+  const registryClient = new ChainRegistryClient();
+  const { ibcConnections } = await registryClient.get(chainId);
+  return ibcConnections;
+};
+
+export const getStakingTokenMetadata = async () => {
+  const chainId = await getChainId();
+  if (!chainId) {
+    throw new Error('Could not fetch chain id');
+  }
+
+  const registryClient = new ChainRegistryClient();
+  const { stakingAssetId } = await registryClient.get(chainId);
+  const stakingAssetsMetadata = await getAssetMetadataById(stakingAssetId);
+
+  if (!stakingAssetsMetadata) {
+    throw new Error('Could not fetch staking asset metadata');
+  }
+  return stakingAssetsMetadata;
+};
```

Comment:
now we get the stakingAssetId from the registry, and the metadata from the view service. (this makes no sense, but we should always favor local data for privacy reasons)

Reply:
Gotcha — perhaps we can add `stakingAssetId` to `AppParameters` in the future.